tests/reset/reset.cpp

```diff
@@ -0,0 +1,467 @@
+    EXPECT_EQ(1, r1.whenCount());
+
+    // Testing move assignment for Reset
+    r2 = std::move(r1);
+    EXPECT_EQ(1, r2.whenCount());
+
+    // Testing move constructor for Reset
+    libcellml::Reset r3 = std::move(r2);
+    EXPECT_EQ(1, r3.whenCount());
+}
+
+TEST(Reset, printResetWithVariable) {
+    std::string e = "<reset variable=\"A\"/>";
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+    libcellml::VariablePtr v = std::make_shared<libcellml::Variable>();
+
+    v->setName("A");
+
+    r->setVariable(v);
+
+    libcellml::Printer p;
+
+    const std::string a = p.printReset(r);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Reset, printResetWithOrder) {
+    std::string e = "<reset order=\"1\"/>";
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+
+    r->setOrder(1);
+
+    libcellml::Printer p;
+
+    const std::string a = p.printReset(r);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Reset, printResetWithOrderAndVariable) {
+    std::string e = "<reset variable=\"B\" order=\"1\"/>";
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+
+    libcellml::VariablePtr v = std::make_shared<libcellml::Variable>();
+
+    v->setName("B");
+
+    r->setVariable(v);
+    r->setOrder(1);
+
+    libcellml::Printer p;
+
+    const std::string a = p.printReset(r);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Reset, printResetWithWhen) {
+    std::string e = "<reset>"
+                        "<when/>"
+                    "</reset>";
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+
+    libcellml::WhenPtr w = std::make_shared<libcellml::When>();
+
+    r->addWhen(w);
+
+    libcellml::Printer p;
+
+    const std::string a = p.printReset(r);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Reset, printResetWithMultipleWhens) {
+    std::string e = "<reset>"
+                        "<when/>"
+                        "<when/>"
+                        "<when/>"
+                    "</reset>";
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+
+    libcellml::WhenPtr w1 = std::make_shared<libcellml::When>();
+    libcellml::WhenPtr w2 = std::make_shared<libcellml::When>();
+    libcellml::WhenPtr w3 = std::make_shared<libcellml::When>();
+
+    r->addWhen(w1);
+    r->addWhen(w2);
+    r->addWhen(w3);
+
+    libcellml::Printer p;
+
+    const std::string a = p.printReset(r);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Reset, printResetWithWhenWithValueSet) {
+    const std::string e = "<reset>"
+                              "<when>"
+                                  "<math xmlns=\"http://www.w3.org/1998/Math/MathML\">"
+                                      "a value set"
+                                  "</math>"
+                              "</when>"
+                          "</reset>";
+
+    libcellml::ResetPtr r = std::make_shared<libcellml::Reset>();
+    libcellml::WhenPtr w = std::make_shared<libcellml::When>();
+
+    w->setValue("<math xmlns=\"http://www.w3.org/1998/Math/MathML\">a value set</math>");
```

Comment:
Am I correct in assuming that all calls to `When::setValue()` will need to include the `math` element? If anything, I would have expected a call like `w->setValue("a value set")`? I mean, the API is supposed to make our life "easy", no?

Reply:
At this time I think this is fine, it wouldn't be hard to add the wrapping if not present if that is desired.

Reply:
Ok for now, but I will certainly like to have such a wrapper.